ios/PolyPodApp/Core/UpdateNotification.swift

```diff
@@ -133,19 +135,32 @@ class UpdateNotification {
     let title = notificationData.localizedTitle
     let text = notificationData.localizedText
     
+    var isProtectedDataAvailable = true
+
     private var cachedState: State
     private var state: State {
         get { cachedState }
         set {
             cachedState = newValue
-            LastUpdateNotification.write(id: id, state: cachedState.rawValue)
+            if isProtectedDataAvailable {
+                LastUpdateNotification.write(id: id, state: cachedState.rawValue)
+            }
         }
     }
     
     init() {
         cachedState = UpdateNotification.loadLastState(id)
     }
     
+    func protectedDataDidBecomeAvailable() {
+        isProtectedDataAvailable = true
+        LastUpdateNotification.write(id: id, state: cachedState.rawValue)
```

Comment:
Let me see if I got this correctly. You write to the LastUpdateNotification in two different paths: if you set the state (not clear when that happens) of when protectedData becomes available. You *never* write if protectedData is not available, that's clear enough. However, not clear what happens in that case. If you change the state but protected data is not available, it's cached but not written. So what's written is not the single source of truth?